### PR TITLE
Feat(Empty States): Handle initial, loading and empty states of list

### PR DESCRIPTION
**Pull request by nemesis09**
Solves issue #42 
 - patternfly-ng Empty State is used to render initial and empty states of the list
 - users-container component now passes an acknowledgement of completion of search to users-list 
   component as a boolean [input] = [isSearchComplete]
 - users-list component implements loading state based on the acknowledgement

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -51,23 +51,26 @@ import { environment } from '../../src/environments/environment';
...
   providers: [
     AuthenticationService,
     Broadcaster,
+    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
+    { provide: HTTP_INTERCEPTORS, useClass: CacheInterceptor, multi: true },
     { provide: AUTH_API_URL, useValue: environment.authApiUrl },
     { provide: SSO_API_URL, useValue: environment.ssoApiUrl },
     { provide: WIT_API_PROXY, useValue: environment.witApiUrl },
     { provide: ADMIN_API_URL, useValue: environment.adminApiUrl },
+    { provide: AUTH_API_URL, useValue: 'https://auth.prod-preview.openshift.io/api/' },
```
Comment (rohitkrai03) [inline src/app/app.module.ts]: @nemesis09 why is this again setting values for api urls directly?

File: src/app/components/users-container/users-container.component.ts [hunk 2]
```diff
@@ -13,16 +13,19 @@ export class UsersContainerComponent implements OnInit, OnDestroy {
   private subscriptions: Subscription = new Subscription();
   users: User[];
   isSubscriptionError: boolean;
+  isSearchComplete: boolean;
```
Comment (rohitkrai03) [inline src/app/components/users-container/users-container.component.ts]: Set default value for the variable. `false`

File: src/app/app.module.ts (hunk 1 above)
Comment (nemesis09) [reply src/app/app.module.ts]: Sorry.
Missed it in cleanup. It's fixed.

File: src/app/components/users-container/users-container.component.ts (hunk 2 above)
Comment (nemesis09) [reply src/app/components/users-container/users-container.component.ts]: okay.
it is initialized to default value in the latest commit.

**Pull request merged**